test/integration/apiserver/oidc/oidc_test.go

```diff
@@ -1789,6 +1898,23 @@ func startTestAPIServerForOIDC[L utilsoidc.JosePublicKey](t *testing.T, c apiSer
 	var customFlags []string
 	if len(c.authenticationConfigYAML) > 0 {
 		customFlags = []string{fmt.Sprintf("--authentication-config=%s", writeTempFile(t, c.authenticationConfigYAML))}
+		// super hacky way to also configure egress selector
+		if strings.Contains(c.authenticationConfigYAML, "### NEEDS EGRESS ON START ###") {
+			udsName := filepath.Join(t.TempDir(), "uds")
+			go runEgressProxy(t, udsName)
```

Comment:
does this make sure the egress proxy is running before we continue / return? will this flake on startup?